flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartIndividualFailoverStrategy.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.flink.runtime.executiongraph.failover.flip1;
+
+import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
+import org.apache.flink.runtime.scheduler.strategy.SchedulingTopology;
+
+import org.apache.flink.shaded.guava18.com.google.common.collect.ImmutableSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+
+public class RestartIndividualFailoverStrategy implements FailoverStrategy {
```

Comment:
As far as I have understood it, this strategy is only usable for failures in sinks (or sink-like nodes) yet, right?
For the final PR, we need to make sure that users do not pick it incorrectly. Maybe it's even too early to introduce it at the current point in time.

Reply:
That's right. This is for end-2-end testing purpose.